custom/icds_reports/management/commands/dashboard_usage_report.py

```diff
@@ -0,0 +1,244 @@
+        print(f'Fetching data from file: {filename}')
+        with open(filename, 'r') as f:
+            reader = csv.reader(f)
+            return [
+                r[0] if len(r) == 1 else r for r in list(reader)
+            ]
+
+
+def _write_to_file(filename, rows):
+    print(f'Writing {len(rows)} to file {filename}')
+    with open(filename, 'w') as f:
+        writer = csv.writer(f)
+        writer.writerows([
+            r if isinstance(r, list) else [r] for r in rows
+        ])
+
+
+class Command(BaseCommand):
+    required_fields = ['state_id', 'state_name', 'district_id', 'district_name', 'block_id', 'block_name']
+    location_types = ['state_id', 'district_id', 'block_id']
+    user_level_list = ['State', 'District', 'Block']
+    location_test_fields = ['state_is_test', 'district_is_test', 'block_is_test', 'supervisor_is_test',
+                            'awc_is_test']
+
+    roles = {
+        '.nod': 'Nodal Officer',
+        '.ncd': 'Consultant(Nutrition & Child Development)',
+        '.bcc': 'Consultant (Behaviour Change Communication & Capacity Building)',
+        '.sdc': 'Consultant (Social Development & Community Mobilization)',
+        '.mne': 'Consultant (Monitoring & Evaluation and Decentralized Planning)',
+        '.fm': 'Consultant (Financial Management)',
+        '.shd': 'Consultant (Procurement) (State Helpdesk)',
+        '.pa': 'Project Associate',
+        '.acc': 'Accountant',
+        '.cta': 'Central Training Agent',
+        '.dhd': 'District Coordinator (District Helpdesk)',
+        '.dpa': 'District Project Assistant',
+        '.dc': 'District Collector',
+        '.bhd': 'Block Coordinator (Block Helpdesk)',
+        '.bpa': 'Block Project Assistant',
+        '.dpo': 'District Programme Officer (DPO)',
+        '.cdpo': 'Child Development Project Officer (CDPO)',
+
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'start_date',
+            type=date_type,
+            help='The start date (inclusive). format YYYY-MM-DD'
+        )
+        parser.add_argument(
+            'end_date',
+            type=date_type,
+            help='The end date (exclusive). format YYYY-MM-DD'
+        )
+
+    def handle(self, start_date, end_date, **options):
+        domain = 'icds-cas'
+
+        print(f'fetching users data')
+        all_usernames = get_all_usernames_by_domain(domain)
+
+        dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
+
+        usernames = [username for username in all_usernames if dashboard_uname_rx.match(username)]
+
+        usage_counts, indicators_count = self.get_dashboard_usage_counts(start_date, end_date, usernames)
+
+        self.get_request_data(usernames, usage_counts, indicators_count)
+        print(f'Request data written to file {REQUEST_DATA_CACHE}')
+        print(f'Request data written to file {CAS_DATA_CACHE}')
+
+    def get_location_id_vs_name_mapping_from_queryset(self, results_queryset):
```

Comment:
may be this wont be required if you directly pull the usage report as mentioned below.

Reply:
done